src/renderer.js

```diff
@@ -73,11 +72,17 @@ function loadTasks() {
 
   // This will be done on loading
   existingTasks.forEach((task) => {
-    const taskDay = parseInt(task.whenTask.substring(8, 10), 10);
-    console.log("taskday: " + taskDay + " today + 1: " + (today + 1));
-    console.log("task.whenTask[6]: " + task.whenTask[6] + " month: " + month);
+    const taskDate = new Date(task.whenTask);
+    const taskDay = taskDate.getDate();
+    const taskMonth = taskDate.getMonth() + 1;
 
-    if (task.whenTask[6] == month && taskDay === today + 1) {
+    const isTomorrow =
+      (taskMonth === month && taskDay - today === 1) || // Tomorrow in the same month
+      (taskMonth - month === 1 &&
+        today === getLastDayOfMonth(month) &&
+        taskDay === 1); // Tomorrow when today is the last day of the month
+
+    if (isTomorrow) {
```

Comment:
Brought out all the logic statements from the if to prevent stacking of statements